### PR TITLE
feat: lastUpdateAt updates on balance changed

### PR DESCRIPTION
**Pull request by yakischik**

### CODE REVIEW

File: erc20/listener/src/main/kotlin/com/rarible/protocol/erc20/listener/service/balance/BalanceReducer.kt [hunk 1]
```diff
@@ -3,32 +3,50 @@ package com.rarible.protocol.erc20.listener.service.balance
...
-        } else {
-            initialBalance
+        val currentData = current.data
+        val currentBalance = currentData.balance
+
+        if (logEvent.status != LogEventStatus.CONFIRMED) {
+            return current.copy(mark = event.mark)
+        }
+
+        val (balance, eventDate) = when (val data = logEvent.data) {
```
Comment (serejke) [inline erc20/listener/src/main/kotlin/com/rarible/protocol/erc20/listener/service/balance/BalanceReducer.kt:31]: just try to cast `logEvent.data` to `Erc20TokenHistory` and take the `data.date` from there (it is the base field of all subclasses listed here), and avoid code duplication and Pairs

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/service/OrderUpdateService.kt [hunk 2]
```diff
@@ -62,26 +66,48 @@ class OrderUpdateService(
         }
     }
 
-    suspend fun updateMakeStock(hash: Word, knownMakeBalance: EthUInt256? = null): Order? =
-        updateMakeStockFull(hash, knownMakeBalance).first
+    suspend fun updateMakeStock(
+        hash: Word,
+        knownMakeBalance: EthUInt256? = null,
```
Comment (serejke) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/service/OrderUpdateService.kt:71]: We can group `knownMakeBalance & updateDate` parameters in newly added class `MakeBalanceState`

File: erc20/listener/src/main/kotlin/com/rarible/protocol/erc20/listener/service/balance/BalanceReducer.kt (hunk 1 above)
Comment (yakischik) [reply erc20/listener/src/main/kotlin/com/rarible/protocol/erc20/listener/service/balance/BalanceReducer.kt]: Fixed

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/service/OrderUpdateService.kt (hunk 2 above)
Comment (yakischik) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/service/OrderUpdateService.kt:71]: There are could be combination data != null and balance == null

**Pull request merged**